addons/hr_holidays/models/hr_leave.py

```diff
@@ -486,11 +486,11 @@ def _compute_department_id(self):
     def _compute_has_stress_day(self):
         date_from, date_to = min(self.mapped('date_from')), max(self.mapped('date_to'))
         if date_from and date_to:
-            stress_days = self.employee_id._get_stress_days(
-                date_from.date(),
-                date_to.date())
-
             for leave in self:
+                stress_days = leave.employee_id._get_stress_days(
+                    date_from.date(),
+                    date_to.date())
+
```

Comment:
`_get_stress_days` is not indicated to work on ensure_one() 🤔 Instead, I think it would be best for that method to work in batch and keep the call out of the loop to avoid multiplying queries